src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPGlobalProperties : IPGlobalProperties
+    {
+        private static readonly string[] s_newLineSeparator = new string[] { Environment.NewLine }; // Used for string splitting
+
+        public override string DhcpScopeName
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override string DomainName
+        {
+            get
+            {
+                return HostInformation.DomainName;
+            }
+        }
+
+        public override string HostName
+        {
+            get
+            {
+                return HostInformation.HostName;
+            }
+        }
+
+        public override bool IsWinsProxy
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override NetBiosNodeType NodeType
+        {
+            get
+            {
+                // Could return NetBiosNodeType.Unknown (?)
+                throw new NotImplementedException();
+            }
+        }
+
+        public override TcpConnectionInformation[] GetActiveTcpConnections()
+        {
+            string tcp4FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp4ConnectionsFile);
+            string[] v4connections = tcp4FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(s_newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
There's a lot of allocation happening in the above four lines.  Is GetActiveTcpConnections something that folks invoke repeatedly?  If so, this would be a good candidate to optimize with some of the helpers we have in Common.